src/capwap_config.erl

```diff
@@ -75,23 +75,21 @@ wtp_static_config(CN) ->
     lager:debug("static config for WTP: ~p", [WTP]),
     WTP.
 
-wtp_config(StaticConfig) ->
-    DefaultEnv = capwap_config:get(wtp, [defaults], []),
-    Defaults = [
-        {psm_idle_timeout,           30},
-        {psm_busy_timeout,           300},
-        {max_stations,               100},
-        {echo_request_interval,      60},
-        {discovery_interval,         20},
-        {idle_timeout,               300},
-        {data_channel_dead_interval, 70},
-        {ac_join_timeout,            70},
-        {admin_pw,                   undefined},
-        {wlan_hold_time,             15},
-        {broken_add_wlan_workaround, false}],
-    DefaultCommon = merge(DefaultEnv, Defaults),
-    WTP = merge(StaticConfig, DefaultCommon),
-    '#new-wtp'(WTP).
+wtp_config(Config) ->
```

Comment:
this is not the right place to handle the merging of the defaults. In fact, the whole API between the generic part and the provider specific part is wrong.

The generic config API is only a very thin wrapper on top the specific providers.
A specific config provider should always return the full configuration for a given WTP. The fact that it might use templates, defaults or other stuff is completely internal to the specific config provider.

Some configuration setting might only be available after a WTP has joined and has send a Configuration Status Request Message. Therefore the API should provide the ability to do a late resolution of those settings.

Rough idea for an API:

```erlang
%% retrieve initial configuration for Join from config store.
-spec load(CN :: binary(), ConfigInformation :: map()) -> ConfigHandle.

%% use to communicate setting form Configuration Update Request to config
-spec update(ConfigHandleIn, ConfigInformation :: map()) -> ConfigHandleOut.

%% get a WTP config setting from config handler
-spec get(ConfigHandle, Selector) -> Values.
```

ConfigHandle is a opaque value as far as the API is concerned, it could be anything or it could already contain the configuration. Point is, the AC does not know and always uses the get function to retrieve the information.